sqlmesh/core/model/definition.py

```diff
@@ -445,6 +445,14 @@ def is_python(self) -> bool:
     def is_seed(self) -> bool:
         return False
 
+    @property
+    def has_self_reference(self) -> bool:
+        return self.name in self._all_model_references
+
+    @property
+    def _all_model_references(self) -> t.Set[str]:
+        return _find_tables(self.render_query())
```

Comment:
this is no longer cached can you remove the _depends_on cache and make a new one for this?